ACTIONS.md

```diff
@@ -0,0 +1,174 @@
+# GitHub Actions setup
+
+Quarkus is built using GitHub actions, with a mix of hosted and self-hosted runners. 
+
+## Setting up a new self-hosted runner on Mac M1 
+
+
+### Non-root user
+
+GitHub actions should not run with administrator privileges. 
+Make an account for the actions runner. *Grant administrator privileges to the account* (we will remove them later).
+
+
+
+### System utilities
+
+As administrator, install homebrew. 
+```shell
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+
+Install build and container utilities:
+
+```bash
+brew install mvn
+brew install gradle
+brew install podman
+```
+
+### Podman setup 
+
+Podman needs some [extra configuration](https://quarkus.io/guides/podman) to work with test containers and dev services. 
+
+Now log in to the account you created to run the actions.
+As the `githubactions` user (but with administrator privileges)
+
+```bash
+PODMAN_VERSION=`podman -v | sed 's/[a-zA-Z ]*//'`
+sudo /opt/homebrew/Cellar/podman/$PODMAN_VERSION/bin/podman-mac-helper install
+```
+The podman helper install seems to be per-user, but it needs to be done with administrator privileges. 
+
+Again as the `githubactions` user, edit ~/.testcontainers.properties and add the following line: `ryuk.container.privileged=true`
+
+```bash
+echo "ryuk.container.privileged=true" >> ~/.testcontainers.properties
+```
+
+```bash
+podman machine set --rootful
+podman machine start
+```
+
+Finally, on ARM, an [extra step](https://edofic.com/posts/2021-09-12-podman-m1-amd64) is required to make AMD64 images work:
+
+```bash
+podman machine ssh
+sudo -i
+rpm-ostree install qemu-user-static
+systemctl reboot
+```
+
+Now remove administrator privileges from the `githubactions` user.
```

Comment:
Maybe an hint of how it can be done? Or it's obvious for someone familiar with macOS?